wave_geometry/include/wave/geometry/exception_helpers.hpp

```diff
@@ -0,0 +1,49 @@
+#ifndef WAVE_EXCEPTION_HELPERS_HPP
+#define WAVE_EXCEPTION_HELPERS_HPP
+
+#include "wave/geometry/rotation.hpp"
+#include "wave/utils/math.hpp"
+
+namespace wave {
+
+
+// This function checks to make sure all elements of the input
+// Eigen array have finite values (not Nan or +/- Inf), as these
+// values are invalid for rotation geometry.
+template <typename MatrixType>
+void checkArrayFinite(const Eigen::MatrixBase<MatrixType> &mat) {
+    if (!mat.allFinite()) {
+        throw std::invalid_argument(
+          "Array input contains non-finite elements.");
+    }
+}
+
+// Function that checks to see if a scalar value is finite,
+// (not Nan or +/- Inf), as these values are invalid for
+// rotation geometry.
+void checkScalarFinite(double input_scalar) {
+    if (!std::isfinite(input_scalar)) {
+        throw std::invalid_argument("Scalar input is non-finite.");
+    }
+}
+
+// Function that checks to see if the input vector is normalized.
+template <typename MatrixType>
+void checkVectorNormalized(const Eigen::MatrixBase<MatrixType> &input_vector) {
+    // fltcmp returns 0 on successful comparision.
+    if (fltcmp(input_vector.norm(), 1)) {
```

Comment:
Not quite ideal:
1. I think you should feed in a specific epsilon (error threshold) yourself, 
2. `fltcmp(input_vector.norm(), 1.0)` if I didn't write this myself I wouldn't know that a return of `0` means "equal", I would have just added `== 0` just to be clear.

Example:
```
void checkVectorNormalized(const Eigen::MatrixBase<MatrixType> &input_vector, double eps=0.0001) {
  // fltcmp returns 0 on if equal
  if (fltcmp(input_vector.norm(), 1.0, eps) != 0) {
```